sdk/storage/Azure.Storage.Blobs/src/BlobDirectoryClient.cs

```diff
@@ -0,0 +1,726 @@
+            ///// <param name="progressHandler">
+            ///// Optional <see cref="IProgress{Long}"/> to provide
+            ///// progress updates about data transfers.
+            ///// </param>
+            //IProgress<long> progressHandler = default,
+            StorageTransferOptions transferOptions = default,
+            CancellationToken cancellationToken = default)
+        {
+            return await StagedDownloadAsync(
+                path,
+                conditions,
+                //progressHandler, // TODO: #8506
+                transferOptions: transferOptions,
+                async: true,
+                cancellationToken: cancellationToken)
+                .ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// This operation will download a blob of arbitrary size by downloading it as individually staged
+        /// partitions if it's larger than the
+        /// <paramref name="transferOptions"/> MaximumTransferLength.
+        /// </summary>
+        /// <param name="path">
+        /// A file path to write the downloaded content to.
+        /// </param>
+        /// <param name="conditions">
+        /// Optional <see cref="BlobRequestConditions"/> to add conditions on
+        /// the creation of this new block blob.
+        /// </param>
+        /// <param name="transferOptions">
+        /// Optional <see cref="StorageTransferOptions"/> to configure
+        /// parallel transfer behavior.
+        /// </param>
+        /// <param name="async">
+        /// Whether to invoke the operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response"/> describing the operation.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        /// TODO: remove static
+        internal static async Task<Response> StagedDownloadAsync(
+            string path,
+            BlobRequestConditions conditions = default,
+            ///// <param name="progressHandler">
+            ///// Optional <see cref="IProgress{Long}"/> to provide
+            ///// progress updates about data transfers.
+            ///// </param>
+            //IProgress<long> progressHandler, // TODO: #8506
+            StorageTransferOptions transferOptions = default,
+            bool async = true,
+            CancellationToken cancellationToken = default)
+        {
+            //TODO: implement directory download
+            BlobBaseClient client = new BlobBaseClient(new Uri("fakeUri"));
+            using Stream destination = File.Create(path);
+            PartitionedDownloader downloader = new PartitionedDownloader(client, transferOptions);
```

Comment:
are expanding partitioneddownloader scope? or should we create DirectoryDownload that would use PArtitionedDownloader for each entry ?
I'd suggest we think about how to break down internal components rather than loading them with extra responsibility. (see [here](https://en.wikipedia.org/wiki/Single-responsibility_principle) )

Reply:
@rnpmsft is currently working on this part to break it down for the directory download and upload.

Reply:
Will be done in a different PR by Rushi